scipy/stats/_stats_py.py

```diff
@@ -4413,6 +4413,9 @@ def pearsonr(x, y, *, alternative='two-sided'):
     x = np.asarray(x)
     y = np.asarray(y)
 
+    if True in np.iscomplex(x) or True in np.iscomplex(y):
```

Comment:
Since we have NumPy arrays, we would usually use `np.any` or the `any` method of the arrays rather than pure-Python.

```suggestion
    if np.iscomplex(x).any() or np.iscomplex(y).any():
```

But that would be slow, because `any` would check every element.

For that reason, I'd suggest checking the dtype:

```suggestion
if (np.issubdtype(x.dtype, np.complexfloating) 
        or np.issubdtype(y.dtype, np.complexfloating)):
```